### PR TITLE
[Feature] Support show create external catalog table

### LINKED ISSUE #7286: Support `show create table` of external catalog
## Feature request

**Is your feature request related to a problem? Please describe.**

https://github.com/StarRocks/starrocks/issues/7268

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by leoyy0316**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #7286

## Problem Summary(Required) ：

support show create external catalog table
**mysql:**
| test_partition | CREATE TABLE jdbc_catalog_01.hive_debug.test_partition(
`id` INT,
`name` VARCHAR(30)
) |

**hudi:**
| hudi_test_par | CREATE TABLE hudi_catalog_03.temp_db.hudi_test_par(
`_hoodie_commit_time` VARCHAR(65533),
`_hoodie_commit_seqno` VARCHAR(65533),
`_hoodie_record_key` VARCHAR(65533),
`_hoodie_partition_path` VARCHAR(65533),
`_hoodie_file_name` VARCHAR(65533),
`id` INT,
`name` VARCHAR(65533),
`ts` BIGINT,
`dt` VARCHAR(65533)
)
WITH (
 partitioned_by = ARRAY[dt]
) |

| hudi_test | CREATE TABLE hudi_catalog_03.temp_db.hudi_test(
`_hoodie_commit_time` VARCHAR(65533),
`_hoodie_commit_seqno` VARCHAR(65533),
`_hoodie_record_key` VARCHAR(65533),
`_hoodie_partition_path` VARCHAR(65533),
`_hoodie_file_name` VARCHAR(65533),
`id` INT,
`name` VARCHAR(65533),
`ts` BIGINT
) |

**hive:**
| test_par | CREATE TABLE hive_catalog.test.test_par(
`id` INT,
`name` VARCHAR(1048576),
`dt` VARCHAR(1048576)
)
WITH (
 partitioned_by = ARRAY[dt]
) |

| test  | CREATE TABLE hive_catalog.test.test(
`id` INT,
`name` VARCHAR(1048576)
) |

**iceberg:**
| Table        | Create Table                                                                  |
+--------------+-------------------------------------------------------------------------------+
| test_iceberg | CREATE TABLE iceberg1.test.test_iceberg(
`name` VARCHAR(65533),
`id` BIGINT
) |

| test_iceberg_par | CREATE TABLE iceberg1.test.test_iceberg_par(
`dt` VARCHAR(65533),
`name` VARCHAR(65533),
`id` BIGINT
)
WITH (
 partitioned_by = ARRAY[dt]
) |

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 36 / 45 (80.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/Table.java|0|2|00.00%|[524, 528]|
|:large_blue_circle:|com/starrocks/catalog/HudiTable.java|0|1|00.00%|[155]|
|:large_blue_circle:|com/starrocks/catalog/IcebergTable.java|0|2|00.00%|[130, 131]|
|:large_blue_circle:|com/starrocks/catalog/DeltaLakeTable.java|0|2|00.00%|[74, 75]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|35|37|94.59%|[696, 700]|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|1|1|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/catalog/Table.java [hunk 1]
```diff
@@ -520,4 +520,12 @@ public Set<MvId> getRelatedMaterializedViews() {
         return relatedMaterializedViews;
     }
 
+    public boolean isUnPartitioned() {
+        return true;
+    }
+
+    public List<String> getPartitionKeys() {
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/catalog/Table.java:527]: you can't define this method. partition keys isn't the same as partition name.

**Comment by leoyy0316** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 39 / 47 (82.98%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/Table.java|0|2|00.00%|[524, 528]|
|:large_blue_circle:|com/starrocks/catalog/IcebergTable.java|0|2|00.00%|[130, 131]|
|:large_blue_circle:|com/starrocks/catalog/DeltaLakeTable.java|0|2|00.00%|[74, 75]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|36|38|94.74%|[701, 705]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/StatisticsCalculator.java|2|2|100.00%|[]|

**Comment by leoyy0316** [comment]
run starrocks_admit_test

**Comment by leoyy0316** [comment]
run starrocks_admit_test

**Comment by leoyy0316** [comment]
run starrocks_admit_test

**Comment by leoyy0316** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 38 / 46 (82.61%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/Table.java|0|2|00.00%|[524, 528]|
|:large_blue_circle:|com/starrocks/catalog/IcebergTable.java|0|2|00.00%|[130, 131]|
|:large_blue_circle:|com/starrocks/catalog/DeltaLakeTable.java|0|2|00.00%|[74, 75]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|35|37|94.59%|[701, 705]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/StatisticsCalculator.java|2|2|100.00%|[]|

**Comment by leoyy0316** [comment]
run starrocks_admit_test

**Comment by leoyy0316** [comment]
run starrocks_admit_test

**Review by stephen-shelby — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by leoyy0316** [comment]
run starrocks_clang-format

**Comment by leoyy0316** [comment]
run pr-title-check

**Comment by leoyy0316** [comment]
run pr-title-check

**Comment by wangyouzhuo** [comment]
LGTM

**Pull request merged**